clients/admin-ui/src/features/custom-fields/CustomFieldModal.tsx

```diff
@@ -0,0 +1,293 @@
+
+  const { data: allowList, isLoading: isLoadingAllowList } =
+    useGetAllowListQuery(customField?.allow_list_id as string, {
+      skip: !customField?.allow_list_id,
+    });
+
+  const [validationSchema, setNewValidationSchema] = useState(
+    optionalValidationSchema
+  );
+
+  if (isLoadingAllowList || !isOpen) {
+    return null;
+  }
+
+  const transformedCustomField = transformCustomField(customField, allowList);
+  const initialValues = transformedCustomField || initialValuesTemplate;
+
+  const handelDropdownChange = (value: FieldTypes) => {
+    if (value === FieldTypes.OPEN_TEXT) {
+      setNewValidationSchema(optionalValidationSchema);
+    } else {
+      setNewValidationSchema(requiredValidationSchema);
+    }
+  };
+
+  // const test = useFormikContext<FormValues>()
+  // test.
+  const handleSubmit = async (
+    values: FormValues
+    // helpers: FormikHelpers<FormValues>
+  ) => {
+    if (
+      [FieldTypes.SINGLE_SELECT, FieldTypes.MULTIPLE_SELECT].includes(
+        values.field_type
+      )
+    ) {
+      const uniqueValues = new Set(
+        values.allow_list?.allowed_values
+          .map((v) => v.toLowerCase().trim())
+          .map((v) => v)
+      );
+      if (uniqueValues.size < values.allow_list!.allowed_values.length) {
+        errorAlert("List item value must be unique");
+        return;
+      }
+
+      const allowListPayload: AllowListUpdate = {
+        ...values.allow_list!,
+      };
+
+      if (values.allow_list_id) {
+        allowListPayload.id = values.allow_list_id;
+      }
+
+      if (!allowListPayload.name) {
+        // The UI no longer surfaces this field. It's a required field in the
+        // backend and must be unique. We generate a random name to avoid collisions.
+        allowListPayload.name =
+          Date.now().toString() + Math.random().toString();
+      }
+
+      const result = await upsertAllowList(allowListPayload);
+      if (!("error" in result) && !values.allow_list_id) {
+        // eslint-disable-next-line no-param-reassign
+        values.allow_list_id = result.data?.id;
+      }
+      // Add error handling here
+    }
+
+    if (values.field_type === FieldTypes.OPEN_TEXT) {
+      // eslint-disable-next-line no-param-reassign
+      values.allow_list_id = undefined;
+    }
+
+    if (
+      [FieldTypes.SINGLE_SELECT, FieldTypes.OPEN_TEXT].includes(
+        values.field_type
+      )
+    ) {
+      // eslint-disable-next-line no-param-reassign
+      values.field_type = AllowedTypes.STRING as unknown as FieldTypes;
+    }
+
+    if (values.field_type === FieldTypes.MULTIPLE_SELECT) {
+      // eslint-disable-next-line no-param-reassign,no-underscore-dangle
+      values.field_type = AllowedTypes.STRING_ as unknown as FieldTypes;
+    }
+
+    const payload = { ...values } as unknown as CustomFieldDefinitionWithId;
+    // @ts-ignore - allow_list needs to be removed from custom field payload
+    delete payload.allow_list;
```

Comment:
keeping the types proper with a `transform` would help here too. normally instead of a `delete` call, you can do a

```typescript
const { allow_list, ...payload } = values;
```

though that is trickier here because of the type casting needed